plugin/core/registry.py

```diff
@@ -29,6 +29,17 @@ def best_session(view: sublime.View, sessions: Iterable[Session], point: Optiona
         return None
 
 
+def sessions(view: sublime.View, capability: Optional[str] = None) -> Generator[Session, None, None]:
+    yield from sessions_for_view(view, capability)
+
+
+def session_by_name(view: sublime.View, session_name: str) -> Optional[Session]:
+    for session in sessions(view):
+        if session.config.name == session_name:
+            return session
+    return None
```

Comment:
I'm aiming to remove these free functions as they can be abused easily. Can you instead add them to LspTextCommand as methods?

Reply:
> Can you instead add them to LspTextCommand as methods?

Unfortunately adding them to `LspTextCommand` wont help me, because I need to get the session  by name in the `DocumentSyncListener` class:
```
class DocumentSyncListener(sublime_plugin.ViewEventListener, AbstractViewListener):
```
as you can see `DocumentSyncListener` doesn't extend from `LspTextCommand`.

I could add the get `session_by_name` as a method on `DocumentSyncListener`, but then I would end up with 2 definitions for `session_by_name` (one in `LspTextCommand`, and the other would be in `DocumentSyncListener` and both would do the same thing, return the session by name, so that seem like code duplication for me). 

> I'm aiming to remove these free functions as they can be abused easily

I am still not sure how those functions could be abused. What is your concern?

Reply:
Oh I did not realize it was needed in DocumentSyncListener :) In that case it's easy to find your session:

```python
for sb in self.session_buffers_async():
    if sb.session.config.name == session_name:
        # ...
```

Reply:
Thanks 🙂

Reply:
1f608ec